lib/manager/pip-compile/artifacts.ts

```diff
@@ -40,6 +40,40 @@ function getPipToolsConstraint(config: UpdateArtifactsConfig): string {
   return '';
 }
 
+const constraintLineRegex = regEx(
+  /^(#.*?\r?\n)+# {4}pip-compile(?<arguments>.*?)\r?\n/
+);
+
+export function constructPipCompileCmd(
+  content: string,
+  inputFileName: string,
+  outputFileName: string
+): string {
+  const headers = constraintLineRegex.exec(content);
+  const args = ['pip-compile'];
+  if (headers) {
+    logger.debug({ header: headers[0] }, 'Found pip-compile header');
+    for (const argument of split(headers.groups.arguments)) {
+      if (['--allow-unsafe', '--generate-hashes'].includes(argument)) {
+        args.push(argument);
+      } else if (argument.startsWith('--output-file=')) {
+        // we don't trust the user-supplied output-file argument; use our value here
+        args.push(`--output-file=${upath.parse(outputFileName).base}`);
```

Comment:
I think we should warn if there is a mismatch. We should at least issue an info log message.